runtime/drivers/snowflake/model_executor_self_objectstore.go

```diff
@@ -0,0 +1,123 @@
+package snowflake
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"net/url"
+
+	"github.com/google/uuid"
+	"github.com/mitchellh/mapstructure"
+	"github.com/rilldata/rill/runtime/drivers"
+	"github.com/rilldata/rill/runtime/drivers/s3"
+)
+
+type selfToObjectStoreExecutor struct {
+	c     *connection
+	store drivers.ObjectStore
+	opts  *drivers.ModelExecutorOptions
+}
+
+var _ drivers.ModelExecutor = &selfToObjectStoreExecutor{}
+
+func (e *selfToObjectStoreExecutor) Execute(ctx context.Context) (*drivers.ModelResult, error) {
+	props := &modelOutputProperties{}
+	if err := mapstructure.Decode(e.opts.OutputProperties, props); err != nil {
+		return nil, err
+	}
+	outputGlob, err := e.export(ctx, e.opts.InputProperties, props.Path)
+	if err != nil {
+		return nil, err
+	}
+	resProps := &modelResultProperties{Path: outputGlob}
+	res := make(map[string]any)
+	err = mapstructure.Decode(resProps, &res)
+	if err != nil {
+		return nil, err
+	}
+
+	return &drivers.ModelResult{
+		Connector:  e.opts.OutputConnector,
+		Properties: res,
+	}, nil
+}
+
+func (e *selfToObjectStoreExecutor) export(ctx context.Context, props map[string]any, outputLocation string) (string, error) {
+	conf := &sourceProperties{}
+	err := mapstructure.Decode(props, conf)
+	if err != nil {
+		return "", err
+	}
+	if conf.SQL == "" {
+		return "", fmt.Errorf("property 'sql' is mandatory for connector \"snowflake\"")
+	}
+
+	var dsn string
+	if conf.DSN != "" { // get from src properties
+		dsn = conf.DSN
+	} else if e.c.configProperties.DSN != "" { // get from driver configs
+		dsn = e.c.configProperties.DSN
+	} else {
+		return "", fmt.Errorf("the property 'dsn' is required for Snowflake. Provide 'dsn' in the YAML properties or pass '--var connector.snowflake.dsn=...' to 'rill start'")
+	}
+
+	db, err := sql.Open("snowflake", dsn)
+	if err != nil {
+		return "", err
+	}
+
+	outputLocation, err = url.JoinPath(outputLocation, "rill-tmp-"+uuid.New().String(), "/")
+	if err != nil {
+		return "", err
+	}
+
+	creds, err := creds(e.store)
+	if err != nil {
+		return "", err
+	}
+
+	//nolint:gosec // can't pass as query args
+	query := fmt.Sprintf(`
+	COPY INTO '%s'
+		FROM (%s) 
+		CREDENTIALS = %s
+		HEADER = TRUE
+		MAX_FILE_SIZE = 536870912 
+		FILE_FORMAT = (TYPE='PARQUET' COMPRESSION = 'SNAPPY')`, outputLocation, conf.SQL, creds)
+	_, err = db.ExecContext(ctx, query)
+	if err != nil {
+		return "", err
+	}
+	return url.JoinPath(outputLocation, "*."+string(drivers.FileFormatParquet))
```

Comment:
So regarding the earlier comment about output properties, I think it would be nicer to return just `outputLocation` here to clearly indicate that the full directory is the model output.

Reply:
The idea was to make sure the `Format` is not fixed and is also returned. With the earlier implementation of `Export` it explicitly returned a `FileFormat`. 
The problem in returning just the `outputLocation` is that clickhouse's integration also requires a glob pattern or file name to work properly. 
Also the `path` parameter for s3 sourceproperties expects it to be a glob.